Automattic/jetpack
# SEO: Add module settings to WP Admin

#### Changes proposed in this Pull Request:

* Make the SEO Tools module settings available in WP Admin; currently WP Admin links to Calypso for these settings.

#### Jetpack product discussion

* Builds upon https://github.com/Automattic/jetpack/pull/17309
* Internal PT: p1HpG7-ba8-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

1. Pull down these changes on a Jetpack connected site.
2. Observe the SEO Tools settings from the Calypso management page at `wordpress.com/marketing/traffic/__SITE_URL__`
3. Make sure the SEO Tools module is turned on `Enable SEO Tools to optimize your site for search engines`
4. Set some custom `Page Title Structure` values for each page-type/section (e.g. Front Page, Archives, etc.); be sure to use some of the buttons to insert the dynamic values (e.g `+ Site Title`, `+ Tagline`, etc.)
5. Navigate to the equivalent WP Admin settings page at `__SITE_URL__/wp-admin/admin.php?page=jetpack#/traffic` 
6. You should see the same values as in Calypso under the `Search engine optimization` section, though placeholders are used here for the dynamic values (e.g. `[site_name]`) in the inputs.
7. You should be able to edit values and save those settings, and see them reflected in Calypso as well.
8. Check the various features such as the `Front Page Meta Description` and live previews.
9. When activating a "conflicting" plugin such as Yoast, the SEO settings will be disabled and a notice shown.

#### Proposed changelog entry for your changes:

SEO Tools: existing settings now available in WP Admin

#### Screenshots:

WP Admin:

![Screen Shot 2021-03-23 at 1 12 17 PM](https://user-images.githubusercontent.com/15803018/112204911-dacd8c00-8bd9-11eb-92ba-578ee3b8ba8a.png)

![Screen Shot 2021-03-23 at 1 12 57 PM](https://user-images.githubusercontent.com/15803018/112204925-ddc87c80-8bd9-11eb-91af-7585f078e152.png)

Original Calypso implementation:

![Screen Shot 2021-03-23 at 1 16 33 PM](https://user-images.githubusercontent.com/15803018/112205128-18cab000-8bda-11eb-9d4f-cc807531f7d0.png)
